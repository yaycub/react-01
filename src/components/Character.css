.characterCard{
  max-width: 300px;
  margin-left: 0.625rem;
  margin-right: 0.625rem;
  margin-bottom: 0.625rem;
  box-shadow: rgba(0, 0, 0, 0.16) 0px 2px 2px 0px, rgba(0, 0, 0, 0.08) 0px 0px 0px 1px;
  border-radius: 0.625rem;
  overflow: hidden;
}

.cardHeader{
  position: relative;
  width: 300px;
  height: 300px;
}

.charImg{
  opacity: 1;
  margin: 0px;
  transition: opacity 0.5s ease 0s;
}

.charName{
  width: 100%;
  opacity: 0.8;
  position: absolute;
  bottom: 0px;
  background: rgb(32, 35, 41);
  padding: 0.625rem;
  color: rgb(245, 245, 245);
  font-size: 1.625rem;
  font-weight: 400;
  font-stretch: expanded;
  margin: 0px;
}

.cardInfo{
  height: 100%;
  color: rgb(255, 152, 0);
  padding: 1.25rem;
  background: rgb(51, 51, 51);
}

.stats{
  display: flex;
  -webkit-box-pack: justify;
  justify-content: space-between;
  -webkit-box-align: center;
  align-items: center;
  flex-wrap: nowrap;
  padding: 0.75rem 0px 0.375rem;
  border-bottom: 1px solid rgb(68, 68, 68);
}

.span{
  font-size: 0.7rem;
  font-weight: 400;
  color: rgb(158, 158, 158);
}

.pStat{
  width: 100%;
  font-size: 0.9rem;
  font-weight: 200;
  text-align: right;
  padding: 0px;
  margin: 0px;
}