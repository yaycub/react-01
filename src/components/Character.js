import React, { Component } from 'react';
import styles from './Character.css';
import PropTypes from 'prop-types';

export default class Character extends Component{
  static propTypes = {
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired,
    lastLoc: PropTypes.string.isRequired
  }

  render(){
    const { image, name, status, species, gender, origin, lastLoc } = this.props;
    return (
      <article className={styles.characterCard}>
        <div className={styles.cardHeader}>
          <img src={image} className={styles.charImg}></img>
          <h2 className={styles.charName}>{name}</h2>
        </div>
        <div className={styles.cardInfo}>
          <div className={styles.stats}>
            <span className={styles.span}>Status</span>
            <p className={styles.pStat}>{status}</p>
          </div>
          <div className={styles.stats}>
            <span className={styles.span}>Species</span>
            <p className={styles.pStat}>{species}</p>
          </div>
          <div className={styles.stats}>
            <span className={styles.span}>Gender</span>
            <p className={styles.pStat}>{gender}</p>
          </div>
          <div className={styles.stats}>
            <span className={styles.span}>Origin</span>
            <p className={styles.pStat}>{origin}</p>
          </div>
          <div className={styles.stats}>
            <span className={styles.span}>Last Location</span>
            <p className={styles.pStat}>{lastLoc}</p>
          </div>
        </div>
      </article>
    );
  }
}
