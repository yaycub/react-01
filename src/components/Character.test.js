import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character Component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character  name="Alan Rails" image="image.jpeg" status="Dead" species="Human, Superhuman (Ghost trains summoner)" gender="Male" origin="unkown" lastLoc="Worldender's lair"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
