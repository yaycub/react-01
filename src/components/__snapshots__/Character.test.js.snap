// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component renders Character 1`] = `
<article
  className="characterCard"
>
  <div
    className="cardHeader"
  >
    <img
      className="charImg"
      src="image.jpeg"
    />
    <h2
      className="charName"
    >
      Alan Rails
    </h2>
  </div>
  <div
    className="cardInfo"
  >
    <div
      className="stats"
    >
      <span
        className="span"
      >
        Status
      </span>
      <p
        className="pStat"
      >
        Dead
      </p>
    </div>
    <div
      className="stats"
    >
      <span
        className="span"
      >
        Species
      </span>
      <p
        className="pStat"
      >
        Human, Superhuman (Ghost trains summoner)
      </p>
    </div>
    <div
      className="stats"
    >
      <span
        className="span"
      >
        Gender
      </span>
      <p
        className="pStat"
      >
        Male
      </p>
    </div>
    <div
      className="stats"
    >
      <span
        className="span"
      >
        Origin
      </span>
      <p
        className="pStat"
      >
        unkown
      </p>
    </div>
    <div
      className="stats"
    >
      <span
        className="span"
      >
        Last Location
      </span>
      <p
        className="pStat"
      >
        Worldender's lair
      </p>
    </div>
  </div>
</article>
`;
